scripts/shmesa/shmesa

```diff
@@ -0,0 +1,455 @@
+    fi
+    
+    if [[ -f history_columns.list ]]; then
+        backup_copy history_columns.list
+    else
+        cp "$MESA_DIR/star/defaults/history_columns.list" .
+    fi
+    
+    local ESCAPE="s#[^^]#[&]#g; s#\^#\\^#g" # sed escape string
+    # Uncomment the specified parameters
+    while [[ -n $1 ]]; do
+        local param=$1
+        shift
+        
+        local escapedParam=$(sed "$ESCAPE" <<< "$param")
+        local search="^\s*\!*\s*$escapedParam\s*.+$"
+        local replace="    $param"
+        
+        if grep -q "$param" profile_columns.list || \
+           grep -q "$param" history_columns.list; then
+            sed -r -i -e "s#$search#$replace#g" profile_columns.list
+            sed -r -i -e "s#$search#$replace#g" history_columns.list
+        else
+            echo "Warning: Parameter '$param' not found in either profile_columns.list or history_columns.list."
+        fi
+    done
+}
+
+
+shmesa_cp () {
+    local USAGE="Usage: shmesa cp source_dir target_dir"
+    if shmesa_check_h_flag "$@"; then
+        echo $usage
+        echo "Copies a MESA working directory but without copying"
+        echo "LOGS, photos, or .mesa_temp_cache"
+        return 0
+    fi
+    
+    if [[ -z $1 || -z $2 ]]; then
+        echo "Error: Missing arguments."
+        echo $usage
+        return 1
+    fi
+    
+    # args: ($1) source directory to be copied from
+    #       ($2) target directory to be copied to
+    local SOURCE=$1
+    local TARGET=$2
+    rsync -av --progress $SOURCE/ $TARGET \
+        --exclude LOGS \
+        --exclude photos \
+        --exclude .mesa_temp_cache \
+        --exclude *.bak
+}
+
+
+shmesa_grep () {
+    local usage="Usage: shmesa grep term [optional: directory or filename]"
+    if shmesa_check_h_flag "$@"; then
+        echo $usage
+        return 0
+    fi
+    
+    if [[ -z $1 ]]; then
+        echo "Error: Missing search term."
+        echo $usage
+        return 1
+    fi
+    
+    local search_term=$1
+    local search_dir=$MESA_DIR
+    
+    if [[ -n $2 ]]; then
+        search_dir=$MESA_DIR/$2
+    fi
+    
+    grep -r --color=always --include \*.f --include \*.f90 --include \*.for \
+        "$search_term" "$search_dir"
+}
+
+
+shmesa_zip () {
+    if shmesa_check_h_flag "$@"; then
+        echo "Usage: shmesa zip [directory]"
+        echo "zips the inlists and models of the specified directory for sharing"
+        return 0
+    fi
+
+    local dir_to_zip='.'
+    local zip_name="mesa.zip"
+    if [[ ! -z $1 ]]; then
+        dir_to_zip=$1
+        zip_name="${dir_to_zip}_mesa.zip"
+    fi
+
+    if [[ ! $(find "$dir_to_zip" -type f -iname "*inlist*" -o -iname "*.mod") ]]; then
+        echo "Warning: No files with 'inlist' in their names or files ending in '.mod' found in the directory '$dir_to_zip'."
+    fi
+
+    # Create a zip file containing only inlists and models
+    zip -r "$zip_name" "$dir_to_zip" -i '*inlist*' '*.mod'
```

Comment:
Can we include `src/` directories here too so that `run_star_extras` get bundled up too? And maybe some other files like `*.list` and `*.net` would be helpful as well?

In playing with this just now, I also noticed that `make/run_star.mod` and `make/run_star_extras.mod` got picked up by this, but the mod files are fortran objects that we don't want to share. Is there an easy way to exclude those?